README.md

```diff
@@ -1,45 +1,91 @@
+ * Code auto-completion - again a standard that you got used to from your desktop tool
+ * Support for number of variables and libraries within the context of the execution
+ * Support latest ECMAScript standard
+ * Run and troubleshoot within the designer interface.
+
+## Available Variables and Libraries
+Here are the available variables and libraries that can be used within the context of execution. The most up-to-date list
+can always be found in be used within the context of execution. The most up-to-date list can always be found in code.js
+of the component. Below is a sample for the reference:
+
+`console`: - more on [Node.js console](https://nodejs.org/dist/latest-v5.x/docs/api/console.html),
+`process`: - Current Node.js process,
+`require`: - Module require,
+`setTimeout`: - more on [setTimeout](https://nodejs.org/dist/latest-v5.x/docs/api/timers.html),
+`clearTimeout`: - more on [clearTimeout](https://nodejs.org/dist/latest-v5.x/docs/api/timers.html),
+`setInterval`: - more on setInterval,
+`clearInterval`: - more on clearInterval,
+`msg`: - Incoming message containing the payload from the previous step,
+`exports`: {},
+`messages`: - Utility for convenient message creation,
+`request`: - Http Client,
+`wait`: - wait,
+`emitter`: user to emit messages and errors
+
+## Code component usage Examples
+
+Use code is very simple, just do following:
+
+```JavaScript
+async function run(msg) {
+  console.log('Incoming message is %s', JSON.stringify(msg));
+  const body = { result : 'Hello world!' };
+  // You can emit as many data messages as required
+  await this.emit('data', { body });
+  console.log('Execution finished');
+}
+```
 
-> If you fail to upload you SSH Key you will get **permission denied** error during the deployment.
+Please note if you have a simple one-in-one-out function you can simply return a JSON object as a result
+of your function, it will be automatically emitted as data.
 
-## Getting Started
+## Common usage scenarios
 
-After registration and uploading of your SSH Key you can proceed to deploy it into our system. At this stage we suggest you to:
-* [Create a team](http://go2.elastic.io/manage-teams) to work on your new component. This is not required but will be automatically created using random naming by our system so we suggest you name your team accordingly.
-* [Create a repository](http://go2.elastic.io/manage-repositories) where your new component is going to *reside* inside the team that you have just created. For a simplicity you can name your repository **code-component** or **code*.
+### Doing complex data transformation
 
-```bash
-$ git clone https://github.com/elasticio/code-component.git code-component
+[JSONata](http://jsonata.org/) is great however sometimes it's easier to do things in JavaScript, if you want to transorm
+an incoming message with code, just use following sample:
 
-$ cd code-component
+```JavaScript
+async function run(msg) => {
```

Comment:
Did you verify that this works?  My understanding is that the current version of this component does not support the `async` keyword.

Reply:
I did